bodhi/server/models.py

```diff
@@ -621,24 +621,74 @@ class RpmPackage(Package):
 
 
 class Build(Base):
+    """
+    This model represents a specific build of a package.
+
+    This model uses single-table inheritance to allow for different build types.
+
+    Attributes:
+        inherited (bool): The purpose of this column is unknown, and it appears to be unused. At the
+            time of this writing, there are 112,234 records with inherited set to False and 0 with
+            it set to True in the Fedora Bodhi deployment.
+        package_id (int): A foreign key to the Package that this Build is part of.
+        release_id (int): A foreign key to the Release that this Build is part of.
+        signed (bool): If True, this package has been signed by robosignatory. If False, it has not
+            been signed yet.
+        update_id (int): A foreign key to the Update that this Build is part of.
+        release (sqlalchemy.orm.relationship): A relationship to the Release that this build is part
+            of.
+        type (int): The polymorphic identify of the row. This is used by sqlalchemy to identify
+            which subclass of Build to use.
+    """
     __tablename__ = 'builds'
     __exclude_columns__ = ('id', 'package', 'package_id', 'release',
                            'release_id', 'update_id', 'update', 'inherited',
                            'override')
-    __get_by__ = ('nvr',)
 
-    nvr = Column(Unicode(100), unique=True, nullable=False)
-    epoch = Column(Integer, default=0)
     inherited = Column(Boolean, default=False)
     package_id = Column(Integer, ForeignKey('packages.id'))
     release_id = Column(Integer, ForeignKey('releases.id'))
-    update_id = Column(Integer, ForeignKey('updates.id'))
     signed = Column(Boolean, default=False, nullable=False)
+    update_id = Column(Integer, ForeignKey('updates.id'))
 
     release = relationship('Release', backref='builds', lazy=False)
 
+    type = Column(Integer, nullable=False)
+    __mapper_args__ = {
+        'polymorphic_on': type,
+        'polymorphic_identity': 0,
+    }
+
+
+class RpmBuild(Build):
+    """
+    Represents an RPM build.
+
+    Note that this model uses single-table inheritance with its Build superclass.
+
+    Attributes:
+        nvr (unicode): A dash (-) separated string of an RPM's name, version, and release (e.g.
+            u'bodhi-2.5.0-1.fc26')
+        epoch (int): The RPM's epoch.
+    """
+
+    __get_by__ = ('nvr',)
+
+    nvr = Column(Unicode(100), unique=True, nullable=False)
```

Comment:
From the docs, "the columns which correspond to a single subclass are nullable" for single-table inheritance. This might work alright at the moment, but I suspect if you were to define a third class and it didn't have a ``nvr`` column the database would throw a fit since this column is not marked as nullable.

You can enforce the non-nullability of the column in a validator, but I don't think you can have the database enforce it.